examples/rquickjs-cli/src/main.rs

```diff
@@ -0,0 +1,42 @@
+use std::io::{stdout, Write};
+
+use rquickjs::{Context, Function, Object, Result, Runtime, Value};
+
+fn print(s: String) {
+    println!("{s}");
+}
+
+fn main() -> Result<()> {
+    let rt = Runtime::new()?;
+    let ctx = Context::full(&rt)?;
+
+    ctx.with(|ctx| -> Result<()> {
+        let global = ctx.globals();
+        global.set(
+            "print",
+            Function::new(ctx.clone(), print)?.with_name("print")?,
+        )?;
+        ctx.eval::<(), _>(
+            r#"
+globalThis.console = {
+  log(v){
+    globalThis.print(`${v}`);
+  }
+}
+"#,
+        )?;
+
+        let console: Object = global.get("console")?;
+        let js_log: Function = console.get("log")?;
+        loop {
+            let mut input = String::new();
+            let _ = stdout().write(b"> ")?;
+            stdout().flush()?;
```

Comment:
Why manually lock stdout? `print!("> ")` should be sufficient here.

Reply:
If flush is not used here to force the output of the buffer content, the prompt will not be display.